docs/sources/reference/components/loki/loki.source.syslog.md

```diff
@@ -125,6 +130,38 @@ The `rfc3164_default_to_current_year` argument is only relevant when `use_incomi
 `rfc3164` message timestamps don't contain a year, and this component's default behavior is to mimic Promtail behavior and leave the year as 0.
 Setting `rfc3164_default_to_current_year` to `true` sets the year of the incoming timestamp to the current year using the local time of the {{< param "PRODUCT_NAME" >}} instance.
 
+{{< admonition type="note" >}}
+The `rfc3164_default_to_current_year`, `use_incoming_timestamp` and `use_rfc5424_message` fields cannot be used when `syslog_format` is set to `raw`.
+{{< /admonition >}}
+
+#### Supported formats
+
+* **`rfc3164`**
+  A legacy syslog format, also known as BSD syslog.
+  Example: `<34>Oct 11 22:14:15 my-server-01 sshd[1234]: Failed password for root from 192.168.1.10 port 22 ssh2`
+* **`rfc5424`**
+  A modern, structured syslog format. Uses ISO 8601 for timestamps.
+  Example: `<165>1 2025-12-18T00:33:00Z web01 nginx - - [audit@123 id="456"] Login failed`.
+* **`raw`**
+  Disables log line parsing. This format allows receiving non-RFC5424 compliant logs, such as [CEF][cef].
+  Raw logs can be forwarded to [`loki.process`](./loki.process.md) component for parsing.
+
+[cef]: https://www.splunk.com/en_us/blog/learn/common-event-format-cef.html
+
+### `raw_format_options`
+
+The `raw_format_options` block configures the `raw` syslog format behavior.
+
+{{< admonition type="note" >}}
+This block can only be used when you set `syslog_format` to `raw`.
+{{< /admonition >}}
+
+The following argument is supported:
+
+| Name                            | Type   | Description                                                                 | Default | Required |
+|---------------------------------|--------|-----------------------------------------------------------------------------|---------|----------|
+| `use_null_terminator_delimiter` | `bool` | Use null-terminator (`\0`) instead of line break (`\n`) to split log lines. | `false` | no       |
```

Comment:
Nit: Are these the only options that make sense, or should we turn this into a string that defaults to `\n`? I'm not sure what syslog protocol typically uses.

Reply:
that sounds good!